source/extensions/filters/http/custom_response/policies/local_response_policy.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "envoy/extensions/filters/http/custom_response/v3/policies.pb.h"
+#include "envoy/extensions/filters/http/custom_response/v3/policies.pb.validate.h"
+
+#include "source/common/router/header_parser.h"
+#include "source/extensions/filters/http/custom_response/policy.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CustomResponse {
+
+class CustomResponseFilter;
+
+class LocalResponsePolicy : public Policy, public Logger::Loggable<Logger::Id::filter> {
+
+public:
+  LocalResponsePolicy(
+      const envoy::extensions::filters::http::custom_response::v3::LocalResponsePolicy& config,
+      Server::Configuration::CommonFactoryContext& context);
+
+  Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap&, bool,
+                                          CustomResponseFilter&) const override;
+
+  Http::Code getStatusCodeForLocalReply(const Http::ResponseHeaderMap& response_headers) const;
+
+private:
+  // Rewrite the response body for locally specified bodies.
+  void formatBody(const Http::RequestHeaderMap& request_headers,
+                  const Http::ResponseHeaderMap& response_headers,
+                  const StreamInfo::StreamInfo& stream_info, std::string& body) const;
+
+  // Body read from local data source.
+  absl::optional<std::string> local_body_;
```

Comment:
Can you mark members of the objects that are created from config `const` please? Since you are sharing config object among many filter instances in different worker threads, it would be good to express that the state in these objects is immutable.

Reply:
Great catch. FWIW, the policies themselves are immutable and the filter only has const access to them.